bindings/README.md

```diff
@@ -0,0 +1,235 @@
+Author: Paul Fitzpatrick
+CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+```
+
+# SWIG-YARP interfaces
+
+This directory is for producing SWIG interfaces to YARP.
+
+This lets you run YARP from Java, Python, Perl, Tcl, Chicken, C#, Ruby,
+Matlab.
+
+Note that while we try to preserve YARP as close to its original form
+as possible, some classes and methods had to be changed because
+of issues with templates and pointers.  So the API is slightly 
+different from the original.  The two ways to find out about the API
+are:
+
+- Compile the java or python interfaces, and then run javadoc
+   or pydoc to build a browsable form of the API.
+
+- Read yarp.i.  This is a bit complicated, but powerful because you'll
+   learn how to fix things if you run into problems.  See www.swig.org
+   for documentation on the syntax of this file.
+
+
+**Note**: all the `cmake` and `make` commands are referring to the **main** YARP build directory.
+
+## Java INTERFACE
+
+To create the Java interface to YARP from linux
+(the process on windows is analogous using cmake):
+
+```
+  ccmake .
+```
+
+Enable the `CREATE_JAVA` option, then
+
+```
+make
+```
+
+If you install, you end up with a shared library called `yarp_java` in
+`${CMAKE_INSTALL_PREFIX}/lib/jni`
+and a `yarp.jar` file (actually it is a symbolic link to a versioned `.jar` in the same folder)
+in `${CMAKE_INSTALL_PREFIX}/share/yarp/java`
+where `${CMAKE_INSTALL_PREFIX}` is the path to your YARP installation.
+An additional library, called `yarp_matlab_java.jar` is created, containing utilities to load YARP 
+classes inside MATLAB.
+
+You should add `${CMAKE_INSTALL_PREFIX}/share/yarp/java` to your Java ClassPath and
+`${CMAKE_INSTALL_PREFIX}/lib/jni` to the `java.library.path` Java setting.
+
+Remember in your Java code, before calling any YARP classes to load the JNI library (usually in a `static` block):
+
+```java
+static {
+    try {
+        System.loadLibrary("yarp_java");
+    } catch (UnsatisfiedLinkError ule) {
+        ule.printStackTrace();
+    } catch (SecurityException se) {
+        se.printStackTrace();
+    } catch (NullPointerException npe) {
+        npe.printStackTrace();
+    } 
+}
+```
+
+
+#### Note from Lorenzo Natale 30/06/2011
+
+If you select `CMAKE_BULID_TYPE="Release"` on Linux
+you may get runtime errors from Java later on. I guess this is related to Sebastien note 
+below.
+
+```
+  "... we had some problems due to a segmentation fault in
+  libjyarp.so (in some generated JNI function) using SWIG 1.3.35 and gcc
+  (Ubuntu 4.3.2-1ubuntu12). I'm not sure whether I missed something, but
+  adding -fno-strict-aliasing to the compile flags in our build system
+  solved the problem."
+```
+
+## PYTHON, PERL, TCL, RUBY INTERFACES
+
+Run:
+
+```
+ccmake .
+```
+  
+And set `CREATE_PERL` and/or `CREATE_PYTHON` and/or `CREATE_TCL` to true.
+Run:
+
+```
+make
+```
+
+You should now have libraries called 
+
+- libyarp.so 
```

Comment:
What is the rationale for this README? ? I guess cleaning it up and merging it to https://github.com/robotology/yarp/blob/master/doc/yarp_swig.dox is a big job and outside this PR, but I am a bit afraid that cleaning it up only from the markdown point of view will lead users to believe that the documentation is now up to date.

Reply:
I simply updated the old readme updating also the Java part (which should now be updated).
Of course it is possible to move everything in the dox and remove the readme file

Reply:
I did not understand that the Java part was updated, I will open a new issue about merging the readme in the doxygen documentation.